### ISSUE TITLE
Release 4.3.10 - Release Candidate 1 - Integration tests

### ISSUE THREAD

**Issue by jmv74211** [wazuh/wazuh#15376]
# Description

| Wazuh QA: Branch | Wazuh QA: Commit | Wazuh: Tag | Wazuh: Commit |
|:--:|:--:|:--:|:--:|
| `4.3` | https://github.com/wazuh/wazuh-qa/commit/64dc406b7fce8f9929479370c988f00bd65c635e | `v4.3.10-rc1` | https://github.com/wazuh/wazuh/commit/89530f11c9e592cd2e551432209b0080f08ff8e5  |

We are going to check that the integration tests of the `4.3` and `master` branch of `wazuh-qa` work correctly using the `4.3.10-rc1` version of `wazuh`.

The tests will be performed both in the local environment and in Jenkins using `CentOS` as the manager OS. As for the agents, `Linux`, `Windows` and `macOS` will be used as required.

#### Main RC issue

- https://github.com/wazuh/wazuh/issues/15372

#### Previous RC analysis

- None. This is the first.

#### References
|Color|Status |
|:--:|:--|
|🟢|All tests passed successfully|
|🟡|All tests passed but there are some warnings, xfails or xpassed|
|🔴|Some tests have failures or errors|
|🔵|Test execution in progress|
|⚫|To Do|
|🟠|Tests failed and passed after relaunching it|
|:purple_circle:| All skipped |

## Test Integration - Results
<table>
<thead>
<tr>
<th style="width: 175px;">Name</th>
<th style="width: 499px;" colspan="6">Jenkins</th>
</tr>
</thead>
<tbody>
<tr>
<td style="width: 175px;">OS</td>
<td style="width: 208px;" colspan="2">Linux</td>
<td style="width: 79px;">Windows</td>
<td style="width: 97px;">Solaris</td>
<td style="width: 97px;" colspan="2">macOS</td>
</tr>
<tr>
<td style="width: 175px;">Target</td>
<td style="width: 103px;">Manager</td>
<td style="width: 99px;">Agent</td>
<td style="width: 79px;">Agent</td>
<td style="width: 97px;">Agent</td>
<td style="width: 97px;" colspan="2">Agent</td>
</tr>
<tr>
<td style="width: 175px;"><strong>active_response</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33817/">🟢</a></td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33818/">🟢</a></td>
<td style="width: 79px;"><a href="https://ci.wazuh.info/job/Test_integration/33818/">🟢</a></td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>agentd</strong></td>
<td style="width: 103px;">NA</td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33819/">🟢</a></td>
<td style="width: 79px;"><a href="https://ci.wazuh.info/job/Test_integration/33819/">🟢</a></td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>analysisd</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33820/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>api</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33821/">🟡</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>authd</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33822/">🟢</a></td>
<td style="width: 99px;">&nbsp;NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>enrollment<br/></strong></td>
<td style="width: 103px;">NA</td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33823/">🟡</a></td>
<td style="width: 79px;"><a href="https://ci.wazuh.info/job/Test_integration/33823/">🟡</a></td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>fim</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33824/">🟡</a></td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33825/">🟡</a></td>
<td style="width: 79px;"><a href="https://ci.wazuh.info/job/Test_integration/33825/">🟢</a></td>
<td style="width: 97px;"><a href="https://ci.wazuh.info/job/Test_integration/33825/">🟢</a></td>
<td style="width: 97px;" colspan="2"><a href="https://ci.wazuh.info/job/Test_integration/33825/">🟢</a></td>
</tr>
<tr>
<td style="width: 175px;"><strong>gcloud<br/></strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33827/">🔴</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>github<br /></strong></td>
<td style="width: 103px;"> <a href="https://ci.wazuh.info/job/Test_integration/33828/">🟢</a></td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33829/">🟢</a></td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>logcollector</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33830/">🟡</td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33831/">🟡</td>
<td style="width: 79px;"><a href="https://ci.wazuh.info/job/Test_integration/33831/">🟡</td>
<td style="width: 97px;"><a href="https://ci.wazuh.info/job/Test_integration/33831/">🟡</td>
<td style="width: 97px;" colspan="2"><a href="https://ci.wazuh.info/job/Test_integration/33831/">🟡</td>
</tr>
<tr>
<td style="width: 175px;"><strong>logtest<br/></strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33832/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>office365<br /></strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33833/">🟢</a></td>
<td style="width: 99px;"><a href="https://ci.wazuh.info/job/Test_integration/33834/">🟢</a></td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>remoted</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33835/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>rids</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33836/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>rootcheck</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33837/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>vulnerability_detector</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33838/">🟡</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
<tr>
<td style="width: 175px;"><strong>wazuh_db</strong></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/33816/">🟢</a></td>
<td style="width: 99px;">NA</td>
<td style="width: 79px;">NA</td>
<td style="text-align: center; width: 97px;">NA</td>
<td style="text-align: center; width: 97px;" colspan="2">NA</td>
</tr>
</tbody>
</table>

**Comment by jmv74211** [comment]
GCLOUD tests have been found to be failing:

```
=========================== short test summary info ============================
FAILED test_gcloud/test_functionality/test_max_messages.py::test_max_messages[get_configuration0-publish_messages2]
Did not receive expected - INFO - Received and acknowledged x messages
============= 1 failed, 57 passed, 30 skipped in 713.82s (0:11:53) =============
```

It will be re-launched several times to find out if it is a one-time failure or false positive.

**Comment by jmv74211** [comment]
After relaunching the tests several times, it has consistently failed again, now with the following errors:


| R1 | R2 | R3 |
|----|----|----|
| [🔴](https://ci.wazuh.info/job/Test_integration/33857/) | [🔴](https://ci.wazuh.info/job/Test_integration/33858/)| [🔴](https://ci.wazuh.info/job/Test_integration/33859/) |

```
=========================== short test summary info ============================
FAILED test_gcloud/test_functionality/test_logging.py::test_logging[get_configuration3-publish_messages0]
FAILED test_gcloud/test_functionality/test_max_messages.py::test_max_messages[get_configuration0-publish_messages0]
FAILED test_gcloud/test_functionality/test_max_messages.py::test_max_messages[get_configuration0-publish_messages1]
FAILED test_gcloud/test_functionality/test_rules.py::test_rules[get_configuration0]
============= 4 failed, 54 passed, 30 skipped in 801.82s (0:13:21) =============
```

Talking with @CarlosRS9, he has again been requested to relaunch the builds, as after reviewing and studying the case, he can not find the cause or reason why they are failing now when in fact there has been no change in the code of that module for `4.3.10`.

@CarlosRS9 is going to perform the corresponding manual tests to ensure proper operation, while I have relaunched 3 builds again.

**Comment by CarlosRS9** [comment]
I have performed a manual test using `v4.3.10-rc1` and I can confirm that **the module is working normally**.



The manual test consisted of the following:

1. Add 100 new messages to the subscription that we are going to read from
2. Enable the `logall` option in the ossec.conf
3. Apply the configuration in ossec.conf
4. Enable debug mode in the local_internal_options.conf
5. Restart Wazuh, check the ossec.log and the archives.log

The configuration used was as follows:
```
<gcp-pubsub>
    <project_id>wazuh-dev-xxxxxx</project_id>
    <subscription_name>wazuh-framework-subscription</subscription_name>
    <credentials_file>gcp_credentials.json</credentials_file>
    <pull_on_start>yes</pull_on_start>
    <interval>25s</interval>
    <max_messages>100</max_messages>
    <logging>info</logging>
  </gcp-pubsub>
```

Attempting to run the test using the QA subscription resulted in no messages processed as there are no more messages available.


There are no error messages in the ossec.log, the message `Received and acknowledged 100 messages` has been found in the ossec.log and the events have been successfully sent to analysisd.

Example of the contents of archives.log
```
2022 Nov 15 12:02:54 wazuh-master->Wazuh-GCloud {"integration": "gcp", "gcp": {"insertId":"17ud3pue330o4","logName":"projects/wazuh-dev-123456/logs/cloudaudit.googleapis.com%2Fdata_access","protoPayload":{"@type":"type.googleapis.com/google.cloud.audit.AuditLog","authenticationInfo":{"principalEmail":"gcp-iam-admin@wazuh-dev-123456.iam.gserviceaccount.com","principalSubject":"serviceAccount:gcp-iam-admin@wazuh-dev-123456.iam.gserviceaccount.com","serviceAccountKeyName":"//iam.googleapis.com/projects/wazuh-dev-123456/serviceAccounts/gcp-iam-admin@wazuh-dev-123456.iam.gserviceaccount.com/keys/185840559813e66b3aa5150e3e799aeb86e55371"},"authorizationInfo":[{"granted":true,"permission":"iam.serviceAccountKeys.list","resource":"projects/-/serviceAccounts/113054794077313350365","resourceAttributes":{"name":"projects/-/serviceAccounts/113054794077313350365"}}],"methodName":"google.iam.admin.v1.ListServiceAccountKeys","request":{"@type":"type.googleapis.com/google.iam.admin.v1.ListServiceAccountKeysRequest","name":"projects/-/serviceAccounts/test-workspace@wazuh-dev-123456.iam.gserviceaccount.com"},"requestMetadata":{"callerIp":"xx.xx.xx.xx","callerSuppliedUserAgent":"(gzip),gzip(gfe)","destinationAttributes":{},"requestAttributes":{"auth":{},"time":"2022-11-09T18:41:41.325276721Z"}},"resourceName":"projects/-/serviceAccounts/113054794077313350365","response":{"@type":"type.googleapis.com/google.iam.admin.v1.ListServiceAccountKeysResponse"},"serviceName":"iam.googleapis.com","status":{}},"receiveTimestamp":"2022-11-09T18:41:41.452684497Z","resource":{"labels":{"email_id":"test-workspace@wazuh-dev-123456.iam.gserviceaccount.com","project_id":"wazuh-dev-123456","unique_id":"113054794077313350365"},"type":"service_account"},"severity":"INFO","timestamp":"2022-11-09T18:41:41.311495844Z"}}
```

With regard to the errors in the automatic tests, after analysing the results of the previous tests, we can conclude the module was executed correctly, as the output in the `ossec.log` indicates that the execution was carried out with the expected parameters and ended correctly.

Output present in the ossec.log:
```
2022/11/15 08:55:11 wazuh-modulesd:gcp-pubsub[15207] wm_gcp.c:259 at wm_gcp_pubsub_run(): DEBUG: Launching command: wodles/gcloud/gcloud --integration_type pubsub --project wazuh-dev-258815 --subscription_id qa_integration_tests-sub --credentials_file /var/ossec/gcp_credentials.json --max_messages 100 --num_threads 1 --log_level 2
2022/11/15 08:55:31 wazuh-modulesd:gcp-pubsub[15207] wm_gcp.c:317 at wm_gcp_pubsub_run(): INFO: - INFO - Received and acknowledged 0 messages
```

Based on the results obtained it seems that there has been a problem in the integration test when uploading the messages during the execution or that these messages were consumed by a process outside the execution of the tests. It will be necessary to investigate exactly what the reason was.

## To summarise
The module works correctly. It remains to be checked if the test suffers from any problem when attempting to upload the messages to be consumed.

**Comment by jmv74211** [comment]
Thanks @CarlosRS9, after checking that manually the module is working as expected, I have opened the following issue https://github.com/wazuh/wazuh-qa/issues/3593 to investigate why these tests are failing now.

Taking into account this situation, All ITs are working as expected 🟢 .
